metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java

```diff
@@ -355,12 +408,79 @@ public void deleteById(PK id) {
         };
     }
 
+    protected StringBuilder startBaseQuery(String... props) {
+        String alias = getEntityAlias();
+        StringBuilder bldr = new StringBuilder("SELECT ");
+        if (props.length == 0) {
+            bldr.append(alias).append(".* ");
+        } else {
+            boolean start = true;
+            for (String prop : props) {
+                if (! start) {
+                    bldr.append(", ");
+                    start = false;
+                }
+                bldr.append(alias).append(".").append(prop);
+            }
+        }
+        return bldr.append(" FROM [").append(getNodeType(getJcrEntityClass())).append("] AS ").append(getEntityAlias()).append(" ");
+    }
+    
+    protected String getEntityAlias() {
+        return "e";
+    }
+
     protected String sanitizeSystemName(String systemName) {
         return INVALID_SYSTEM_NAME_PATTERN.matcher(systemName).replaceAll("_");
     }
 
     protected String sanitizeTitle(String title) {
         return StringEscapeUtils.escapeJava(title);
     }
+    
+    protected void appendJoins(StringBuilder bldr, String filter) {
+        // No joins by default.  Subclasses should override to add any joins needed for sorting and/or filtering.
+    }
+    
+    protected void appendJoins(StringBuilder bldr, Pageable pageable, String filter) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "pageable". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)